fake_spectra/abstractsnapshot.py

```diff
@@ -302,10 +334,14 @@ def get_data(self, part_type, blockname, segment):
         except bigfile.BigFileError:
             raise KeyError("Not found:"+str(part_type)+"/"+blockname)
 
-    def get_n_segments(self):
-        """Return the number of segments. Number of files on HDF5,
-           but may be whatever in convenient for bigfile."""
-        return int(np.max([1,np.sum(self.get_npart())/(2*256.**3)]))
+    def get_n_segments(self, part_type, chunck_size = 256.**3):
+        """Return the number of segments on each rank that we chop the data into """ 
+        # first get number of particles per rank
+        self.parts_rank = ((self.get_npart()[part_type]//self.size)*np.ones(shape=(self.size,))).astype(int)
+        remainder = int(self.get_npart()[part_type]%self.size)
+        if remainder !=0 :
+            self.parts_rank[0:remainder] += 1
```

Comment:
I don't totally understand this line: could you add a comment?

Reply:
Also chunck -> chunk is the right spelling

Reply:
It is distributing particles among ranks, also divides  the particles on each rank into segments of  2*256**3 . I adopted this segmentation scheme from the previous version, I think it was for less memory  usage, right ?

Are the comments sufficient now ?

Reply:
I see - I was worried about the += 1 and the possibility of some edge cases losing a single particle, but I see now that this is fine because remainder < self.size. Looks good to me now!